### PR TITLE
More API Blueprint fixes

### PR DESCRIPTION
**Pull request by icidasset**
Hi 👋 

Thanks for fixing the API Blueprint stuff.
Here are some further improvements I made (possibly):
- Adds api explanation to the template
- Adds the ability to define explanations for `route`s and methods (eg. `get ...`)
- Properly renders attributes
- Improve parameter/attribute properties (set to "optional" if not required, reverse direction)

Does this fix the issue you have with attributes?

### CODE REVIEW

File: templates/rspec_api_documentation/api_blueprint_index.mustache [hunk 1]
```diff
@@ -16,34 +19,49 @@ FORMAT: 1A
 ## {{ route_name }} [{{ route }}]
 {{# description }}
 
-description: {{ description }}
+{{ description }}
```
Comment (icidasset) [inline templates/rspec_api_documentation/api_blueprint_index.mustache:22]: Not sure what value this should have... or if it even works.

Comment (yld) [reply templates/rspec_api_documentation/api_blueprint_index.mustache:22]: Cool, thanks, we just need to add apiblueprint validations in spec suite, and we'll be done.

Comment (icidasset) [reply templates/rspec_api_documentation/api_blueprint_index.mustache:22]: Fixed the specs, and added an expectation for the "route explanations". I can't seem to write an expectation for the "method explanations", although it works when I use the gem. Something seems off with the `http_method`s in the specs for some reason.

File: templates/rspec_api_documentation/api_blueprint_index.mustache [hunk 2]
```diff
@@ -16,34 +19,49 @@ FORMAT: 1A
...
+
+{{{ explanation }}}
+{{/ explanation }}
 {{# has_attributes? }}
 
-+ Attributes (object)
++ Parameters
+
 {{# attributes }}
-  + {{ name }}{{# example }}: {{ example }}{{/ example }}{{# properties_description }} ({{ properties_description }}){{/ properties_description }}{{# description }} - {{ description }}{{/ description }}
```
Comment (icidasset) [inline templates/rspec_api_documentation/api_blueprint_index.mustache]: `example` is the RSpec example in this case, which we don't want. I wanted to render the actual value of the param here, but I don't think it's possible without some serious refactoring. I did add the ability to define the "default" though.

Anyhow, if you have an idea how we could get the actual parameter value in here that would be amazing 🙏

**Pull request merged**